.github/workflows/check-package-restore.yml

```diff
@@ -0,0 +1,64 @@
+---
+name: Check Package Restore 📦
+on: push
+permissions:
+  contents: read
+jobs:
+  main:
+    name: Check renv restore
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        platform: [linux/amd64, linux/arm64]
+        directory:
+          [
+            "RNA-seq",
+            "basic-teal",
+            "efficacy",
+            "exploratory",
+            "longitudinal",
+            "early-dev",
+            "patient-profile",
+            "python",
+            "safety",
+          ]
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Extract R version from lockfile
+        run: printf 'R_VERSION=%s\n' "$(jq --raw-output .R.Version ${{ matrix.directory }}/renv.lock)" >> $GITHUB_ENV
+
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ env.R_VERSION }}
+
+      - name: Setup system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install --yes \
+            libcurl4-openssl-dev \
+            libgdal-dev \
+            libproj-dev \
+            libudunits2-dev
+
+      - name: Restore renv from cache
+        uses: actions/cache@v2
+        env:
+          CACHE_KEY: renv-${{ runner.arch }}-${{ runner.os }}-${{ env.R_VERSION }}
+        with:
+          path: renv/library
+          key: ${{ env.CACHE_KEY }}-${{ hashFiles('${{ matrix.directory }}/renv.lock') }}
+          restore-keys: ${{ env.CACHE_KEY }}-
+
+      - name: Sync renv with lockfile
+        shell: Rscript {0}
+        run: |
+          setwd("${{ matrix.directory }}")
+          sys.source(".Rprofile")
+          options(renv.config.cache.symlinks = FALSE)
+          options(renv.config.install.remotes = FALSE)
+          renv::restore(clean = TRUE)
```

Comment:
this is quite lenghty command to complete. At the current moment it's 40+ mins and still counting... for multiple apps...

Reply:
Currently, the cache does not seem to work. But once the renv cache works this should not take too long for renv.lock that has no change. I haven't debugged the cause for this yet.

Reply:
So we have 9 apps checked on 2 platforms which gives us 18 workflows to be run. Each of them lasts aproximately ~1h (rounding up) which makes each commit "costs" us 18 hours of CI time. That's way to much.

> This is quite natural for a large shiny app, As far as I know, there is not much we can do about it.

Yes that's true - that's why I am asking. Do we really want to trigger this on each commit? Maybe trigger it on changes in .lock files only? I was even thinking of abandoning this at all and leave it for deploy action to handle as deploy will have to anyway restore lock files.

Reply:
I really like the idea of triggering it on changes to the lock file. Ideally, the current CI file should be cached based on renv.lock hash but for some incorrect config the cache is not working.

I like abandoning this and merging this with the deploy CI. Additionally, we could benefit from running them in series as most of the package across the renv.lock are same.
Regardless we should fix the renv cache issue to speed up the CI runs.

Reply:
I suggest doing now the latter - i.e. remove it complitely and include that in the deploy CI